screens/ChatScreen.js

```diff
@@ -42,126 +44,249 @@ class ChatScreen extends Component {
+      messages: previousState.messages.concat(localMessages),
     }));
   }
 
-  async onReceive(messages) {
-    console.log('onReceive');
-    this.setState(previousState => ({
-      messages: previousState.messages.concat(messages),
-    }));
-    console.log('onReceive done');
+  async createChannelWithUser(chatClient) {
+    console.log('Creating channel...');
+
+    const user = this.props.navigation.state.params.mentee._id;
+    // Sort the IDs so that we have a deterministic order
+    const channelName = this.getChannelName();
+    return chatClient.createChannel({
+      uniqueName: channelName,
+      friendlyName: channelName,
+    }).then((channel) => {
+      console.log('Created channel');
+      channel.join().catch((err) => {
+        console.error(
+          `Couldn't join channel ${channel.friendlyName} because ${err}`
+        );
+      });
+
+      // Invite other user to your channel
+      channel.invite(user).then(() => {
+        console.log('Your friend has been invited!');
+      }).catch((error) => {
+        console.log(`Couldn't invite user: ${error}`);
+      });
+    }).catch((error) => {
+      console.log(`Error in creating channel: ${error}`);
+    });
   }
 
-  getTwilioToken = async (localToken, userId) => {
-    console.log('Getting Twilio data');
-    console.log(localToken);
-    console.log(userId);
-    const API_URL = 'http://ppeheymentor-env.qhsppj9piv.us-east-2.elasticbeanstalk.com';
+  async initChatClient(token) {
+    console.log('initChatClient');
+
+    TwilioChatClient.create(token, { logLevel: 'info' }).then((chatClient) => {
+      this.client = chatClient;
+
+      this.client.on('tokenAboutToExpire', () => {
+        this.getTwilioToken()
+          .then(newData => this.client.updateToken(newData))
+          .catch((err) => {
+            console.log(`Error getting token on refresh: ${err}`);
+          });
+      });
+
+      // Listen for new invitations to your Client
+      this.client.on('channelInvited', (channel) => {
+        console.log(`Invited to channel ${channel.friendlyName}`);
+        // Join the channel that you were invited to
+        // TODO: add a call to the backend to check if we should be joining this channel
+        console.log('Joining channel');
+        channel.join();
+      });
+
+      this.client.on('channelJoined', (channel) => {
+        console.log(`Joined channel ${channel.friendlyName}`);
+      });
+
+      this.channel = this.getChannelForChat(this.client);
+      this.getMessage();
+      // this.subscribeToAllChatClientEvents(); TODO: getting logging failures here
+    }).catch((error) => {
+      console.log('Error while trying to create Twilio client');
+      console.log(error);
+    });
+  }
 
-    console.log(`${API_URL}/chat/${userId}?token=${localToken.api_key}`);
+  async subscribeToAllChatClientEvents() {
+    this.client.on('tokenAboutToExpire', obj => console.log('ChatClientHelper.client', 'tokenAboutToExpire', obj));
```

Comment:
Should we be logging sensitive tokens?

Reply:
We shouldn't be. Right now we don't call this function, I have it here as a reminder for the functions that the client should be handling. I'm not certain yet which ones we will want moving forward. Added a TODO item to decide and clean up.